services/ingestor/compose.yaml

```
@@ -55,31 +55,44 @@ configs:
-          # - epu="xmls/"
-          Methods:
-            - Name: Single Particle
-              Schema: oscem_schemas_spa.schema.json
-              Url: https://raw.githubusercontent.com/osc-em/OSCEM_Schemas/refs/heads/any_fix/project/spa/jsonschema/oscem_schemas_spa.schema.json
-            - Name: Cellular Tomography
-              Schema: oscem_cellular_tomo.json
-              Url: https://raw.githubusercontent.com/osc-em/OSCEM_Schemas/refs/heads/any_fix/project/cellular_tomo/jsonschema/oscem_schemas_cellular_tomo.schema.json
-            - Name: Tomography
-              Schema: oscem_tomo.json
-              Url: https://raw.githubusercontent.com/osc-em/OSCEM_Schemas/refs/heads/any_fix/project/subtomo/jsonschema/oscem_schemas_subtomo.schema.json
-            - Name: EnvironmentalTomography
-              Schema: oscem_env_tomo.json
-              Url: https://raw.githubusercontent.com/osc-em/OSCEM_Schemas/refs/heads/any_fix/project/env_tomo/jsonschema/oscem_schemas_env_tomo.schema.json
+          - Name: LS
+            GithubOrg: SwissOpenEM
+            GithubProject: LS_Metadata_reader
+            Version: v2.0.1
+            Executable: LS_Metadata_reader
+            Checksum: 83ae1b2d469cec10fdcc3ad7cb6c824a389e501e7142417792f2ac836ab174c3
+            ChecksumAlg: sha256
+            CommandLineTemplate: "-i '{{.SourceFolder}}' -o '{{.OutputFile}}' --cs 2.7 --folder_filter=^.*(?i)(metadata).*$"
+            AdditionalParameters:
+            - Param1=2.7
+            - Param2="None"
+            - Param3="xmls/"
+            Methods:
+              - Name: Single Particle
+                Schema: oscem_schemas_spa.schema.json
+                Url: https://raw.githubusercontent.com/osc-em/OSCEM_Schemas/refs/heads/main/project/spa/jsonschema/oscem_schemas_spa.schema.json
+              - Name: Cellular Tomography
+                Schema: oscem_cellular_tomo.json
+                Url: https://raw.githubusercontent.com/osc-em/OSCEM_Schemas/refs/heads/main/project/cellular_tomo/jsonschema/oscem_schemas_cellular_tomo.schema.json
+              - Name: Tomography
+                Schema: oscem_tomo.json
+                Url: https://raw.githubusercontent.com/osc-em/OSCEM_Schemas/refs/heads/main/project/subtomo/jsonschema/oscem_schemas_subtomo.schema.json
+              - Name: EnvironmentalTomography
+                Schema: oscem_env_tomo.json
+                Url: https://raw.githubusercontent.com/osc-em/OSCEM_Schemas/refs/heads/main/project/env_tomo/jsonschema/oscem_schemas_env_tomo.schema.json
+
+          - Name: MS
+            GithubOrg: SwissOpenEM
+            GithubProject: MS_Metadata_reader
+            Version: v1.0.0
+            Executable: MS_Metadata_reader
+            Checksum: d5bb07787544dea4db8d02f6276369e9726a098228392001229ae9d9d716cdf9
+            ChecksumAlg: sha256
+            CommandLineTemplate: "'{{.SourceFolder}}' '{{.OutputFile}}'"
+            Methods:
+              - Name: General
```

```suggestion
              - Name: Material Science
```
Good point about the name. While it made sense for the schema to be called 'General', the extractor is specific to some file types.

Maybe the input format should actually be reflected in all the options:
- EPU or SerialEM > Single Particle Analysis
- EPU or SerialEM > Tomography > Cellular Tomography
- EPU or SerialEM > Tomography > Environmental Tomography
- EPU or SerialEM > Tomography > General Tomography
- .emd or .prz > Material Science > General

---

another thing I wanted to know, are we planning to have multiple versions of extractors co-existing? because then we need to identify a verision of extractor in the names of the schema. Otherwise in frontend there might be two same methods.

---

There are no plans for this but it would be supported by giving distinct `Name` fields for each version